### PR TITLE
Refactor oas

### PR DESCRIPTION
**Pull request by niclim**
## 🍗 Description
_What does this PR do? Anything folks should know?_

Small refactor to 
- remove dead code
- use shared segment
- use shared sentry

## 📚 References
_Links to relevant docs (Notion, Twist, GH issues, etc.), if applicable._

## 👹 QA
_How can other humans verify that this PR is correct?_

### CODE REVIEW

File: projects/optic/src/commands/oas/operations/infer-path-structure.ts [hunk 1]
```diff
@@ -323,63 +315,6 @@ export class InferPathStructure {
   };
 }
 
-export async function computeInferredOperations(
```
Comment (niclim) [inline projects/optic/src/commands/oas/operations/infer-path-structure.ts:326]: only used in 1 test (?)

File: projects/optic/src/commands/oas/diffing/document.ts [hunk 2]
```diff
@@ -128,53 +65,60 @@ export function parseAddOperations(
...
+export async function addIfUndocumented(
+  operationsToAdd: ParsedOperation[],
+  isAddAll: boolean,
+  statusObservations: StatusObservations,
+  interactions: CapturedInteractions,
+  spec: OpenAPIV3.Document,
+  sourcemap: SpecFilesSourcemap
+): Promise<Result<RecentlyDocumented, string>> {
+  const operationsToUpdate = isAddAll
+    ? await observationToUndocumented(
```
Comment (niclim) [inline projects/optic/src/commands/oas/diffing/document.ts:77]: consolidated 2 fns into this

File: projects/optic/src/commands/oas/captures/getInteractions.ts [hunk 3]
```diff
@@ -90,11 +89,7 @@ export async function getInteractions(
     for (const path of harPaths) {
       let harFile = fsSync.createReadStream(path);
       let harEntryResults = HarEntries.fromReadable(harFile);
-      let harEntries = AT.unwrapOr(harEntryResults, (err) => {
-        let message = `HAR entry skipped: ${err.message}`;
-        console.warn(message); // warn, skip and keep going
```
Comment (acunniffe) [inline projects/optic/src/commands/oas/captures/getInteractions.ts:95]: 👍

**Review by acunniffe — APPROVED** [review]
Looks good. Seems like the undocumented stuff will change a bit more as we update the design. This is good cleanup before that.

**Pull request merged**